alexa/request/certificate.php

```diff
@@ -0,0 +1,185 @@
+		$cert_key = openssl_pkey_get_public( $this->certificate_content );
+		$signature = base64_decode( $this->request_signature );
+
+		$valid = openssl_verify( $request_data, $signature, $cert_key, self::ENCRYPT_METHOD );
+		if ( ! $valid ) {
+			throw new InvalidArgumentException( 'Request signature could not be verified' );
+		}
+	}
+
+	/**
+	 * Returns true if the ceertificate is not expired.
+	 *
+	 * @param array $parsed_certificate
+	 * @return boolean
+	 */
+	public function validate_certificate_date( array $parsed_certificate ) {
+		$valid_from = $parsed_certificate['validFrom_time_t'];
+		$valid_to = $parsed_certificate['validTo_time_t'];
+		$time = time();
+		return ( $valid_from <= $time && $time <= $valid_to );
+	}
+
+	/**
+	 * Returns true if the configured service domain is present/valid, false if invalid/not present
+	 * @param array $parsed_certificate
+	 * @return boolean
+	 */
+	public function validate_certificate_san( array $parsed_certificate, $amazon_service_domain ) {
+		if ( false === strpos( $parsed_certificate['extensions']['subjectAltName'], $amazon_service_domain ) ) {
+			return false;
+		} else {
+			return true;
+		}
+	}
+
+	/**
+	 * Verify URL of the certificate
+	 * @throws InvalidArgumentException
+	 * @author Emanuele Corradini <emanuele@evensi.com>
+	 */
+	public function verify_signature_certificate_url() {
+		$url = parse_url( $this->certificate_url );
+
+		if ( static::SIGNATURE_VALID_PROTOCOL !== $url['scheme'] ) {
+			throw new InvalidArgumentException( 'Protocol isn\'t secure. Request isn\'t from Alexa.' );
+		} else if ( static::SIGNATURE_VALID_HOSTNAME !== $url['host'] ) {
+			throw new InvalidArgumentException( 'Certificate isn\'t from Amazon. Request isn\'t from Alexa.' );
+		} else if ( 0 !== strpos( $url['path'], static::SIGNATURE_VALID_PATH ) ) {
+			throw new InvalidArgumentException( 'Certificate isn\'t in "' . static::SIGNATURE_VALID_PATH . '" folder. Request isn\'t from Alexa.' );
+		} else if ( isset( $url['port'] ) && static::SIGNATURE_VALID_PORT !== $url['port'] ) {
+			throw new InvalidArgumentException( 'Port isn\'t ' . static::SIGNATURE_VALID_PORT. '. Request isn\'t from Alexa.' );
+		}
+	}
+
+
+	/**
+	 * Parse the X509 certificate
+	 * @param $certificate The certificate contents
+	 */
+	public function parse_certificate( $certificate ) {
+		return openssl_x509_parse( $certificate );
+	}
+
+	/**
+	 * Return the certificate to the underlying code by fetching it from its location. Cached for 1 min
+	 */
+	public function get_certificate() {
+		$certificate_id = 'alexawp' . md5( $this->certificate_url . $this->app_id );
+		if ( ! $certificate = get_transient( $certificate_id ) ) {
+			$certificate = $this->fetch_certificate();
+			set_transient( $certificate_id, $certificate, TIMESTAMP_VALID_TOLERANCE_SECONDS );
+		}
+		return $certificate;
+	}
+
+	/**
+	 * Perform the actual download of the certificate
+	 */
+	public function fetch_certificate() {
```

Comment:
I don't believe WP.com VIP allows curl. I had to rewrite something similar for Apple News. You probably want to see about rewriting this using the WP HTTP functions.